core/modules/comment/comment.module

```diff
@@ -1479,21 +1471,17 @@ function comment_config_create_validate(Config $staging_config, $all_changes) {
  * replied to. This prevents people from changing or revising their statements
  * based on the replies to their posts.
  *
- * @param $op
+ * @param string $op
  *   The operation that is to be performed on the comment. Only 'edit' is
  *   recognized now.
- * @param $comment
+ * @param Comment $comment
  *   The comment object.
  *
- * @return
+ * @return bool
  *   TRUE if the current user has acces to the comment, FALSE otherwise.
  */
-function comment_access($op, $comment) {
-  global $user;
-
-  if ($op == 'edit') {
-    return ($user->uid && $user->uid == $comment->uid && $comment->status == COMMENT_PUBLISHED && user_access('edit own comments')) || user_access('administer comments');
-  }
+function comment_access($op, Comment $comment) {
+  return $comment->access($op);
```

Comment:
This would be a good place to introduce the `'edit'` -> `'update'` backwards compatibility layer. If "edit" is passed, convert it to "update" and log a deprecated notice. You won't be able to call it directly, but check [watchdog_deprecated_function](https://api.backdropcms.org/api/backdrop/core%21includes%21bootstrap.inc/function/watchdog_deprecated_function/1) on how it calls watchdog(). Checking if logging level is set to debug beforehand may also be needed.

Reply:
It also seems like "create" access may be necessary to check here (matching `node_access()`). In which case we couldn't typehint `Comment $comment`. What do you think?

Reply:
The difference between comment and node is that comment doesn't have bundles, where node does. So the function would have to be changed to 
function comment_access($op, Comment $comment = NULL) {}

Reply:
Also, not sure how to write the deprecated message. It seems we would need to add a function to output deprecated parameters for a function.
Any suggestions for a good reusable message?

Reply:
> The difference between comment and node is that comment doesn't have bundles, where node does. So the function would have to be changed to
function comment_access($op, Comment $comment = NULL) {}

Good point. So the parameter could still be type-hinted and `create` could be supported. Seems good.

As for the message, yes either we could add `watchdog_deprecated_parameter()` or you could manually call watchdog as necessary:

```
if (!watchdog_severity_enabled(WATCHDOG_DEPRECATED)) {
  watchdog('comment', 'Deprecated parameter "edit" passed to comment_access(). Use "update" instead.', array(), WATCHDOG_DEPRECATED);
}
```

I'm not sure how easily that could be abstracted. Seems that the message would need to be entirely variable based on the possibilities for the parameters to be changed. It's not always a simply replace X with Y situation.